# Lab 4 - Propagation of Uncertainty II
        May 17, 2018
        
[<< Table of Contents](index.html)

## P4.1


Take the derivative of $f(x) = \sin(x) x^2$. The value of the derrivative indicates the slope of the function.

<sub>4.1</sub>
$$
f'(x) = \frac{d}{d x}\big(\sin(x) x^2\big)
$$
<br>
$$
f'(x) = \bigg(\frac{d}{d x}\sin(x)\bigg) x^2 + \bigg(\frac{d}{d x}x^2\bigg) \sin(x)
$$
<br>
<sub>4.2</sub>
$$
f'(x) = \cos(x)x^2 + 2x\sin(x)
$$

## P4.2

##### a) Evaluate equation 4.2 at `x = 4`, `x = 5.087`, and `x = 5.5` and visually verify that these values are indeed the slope of the function in the following figure:

![Figure 4.1](lab4_fig1.png)

$$
f'(x) = \cos(x)x^2 + 2x\sin(x)
$$
<br><br>
$$
f'(4) = \cos(4)4^2 + 2(4)\sin(4) = -16.51271
$$
<br>

$$
f'(5.087) = \cos(5.087)5.087^2 + 2(5.087)\sin(5.087) = 0.00044
$$
<br>

$$
f'(5.5) = \cos(5.5)5.5^2 + 2(5.5)\sin(5.5) = 13.67631
$$
<br>

These indeed match the graph above.


##### (b) In your lab notebook, write down a few sentences describing what a derivative is. Make a sketch to help you with your explanation.

A derivative is simple to understand. It is the rate of change of a changing rate. In other words, it is the slope of a slope. The figure below (Figure 4.2 from the lab manual) clearly shows that a function with a non-constant slope can have instantaneous slope, which is what a derrivative is measuring.

<img src="lab4_fig2.png" width="300px"/>


## P4.3

<sub>4.3</sub>
$$
f(x,t) = \sin(t) x^2
$$

##### (a) calculate the partial derivative of function 4.3 with respect to $x$

$$
\frac{\delta f}{\delta x} = \frac{\delta}{\delta x}\big(\sin(t) x^2\big)
$$
<br>
$$
\frac{\delta f}{\delta x} = 2x\sin(t)
$$
<br>

##### (b) calculate the partial derivative of function 4.3 with respect to $t$

$$
\frac{\delta f}{\delta t} = \frac{\delta}{\delta t}\big(\sin(t) x^2\big)
$$
<br>
$$
\frac{\delta f}{\delta t} = \cos(t) x^2
$$
<br>

## P4.4

<sub>4.4</sub>
$$
f(x,y) = (x^2 y + y^2 x) \sin(2x-y^2)
$$

##### (a) calculate the partial derivative of function 4.4 with respect to $x$

$$
\frac{\delta f}{\delta x} = \frac{\delta}{\delta x}\big((x^2 y + y^2 x) \sin(2x-y^2)\big)
$$
<br>
$$
\frac{\delta f}{\delta x} = (2xy + y^2)\sin(2x-y^2) + 2(x^2 y + y^2 x)\cos(2x-y^2)
$$
<br>

##### (b) calculate the partial derivative of function 4.4 with respect to $y$

$$
\frac{\delta f}{\delta y} = \frac{\delta}{\delta y}\big((x^2 y + y^2 x) \sin(2x-y^2)\big)
$$
<br>
$$
\frac{\delta f}{\delta y} = (x^2 + 2xy)\sin(2x-y^2) - 2y(x^2 y + y^2 x)\cos(2x-y^2)
$$
<br>

## P4.5 
Consider the following situation. Using a cannon, tilted at an angle $\theta = 35^\circ$,
you launch a cannonball through the air. You’d like to predict where the
ball will land with it’s associated uncertainty. If the ball is launched from
level ground (lands at the same height as it was launched) you can use the
following equation:

<sub>4.16</sub>
$$
x = \frac{v_0^2 \sin(2\theta)}{g}
$$

##### (a) Using the same procedure as last week, calculate the exit speed of the sphere. You may choose the launch-strength setting to be 1-click, 2-clicks, or 3-clicks. Use the standard deviation from last week’s data as your uncertainty on this speed. Record these numbers in your lab notebook.

We launched off the rocket straight up in 5 trials. Below are the height results.

| Trial # | Height (cm)|
|--|
| 1 | 131 |
| 2 | 130 |
| 3 | 131 |
| 4 | 130 |
| 5 | 130 |

In [1]:
# calculate the average v0
import numpy as np
heights = [131,130,131,130,130]
uncertainty= 0.08 # last weeks std
g = 9.81 # m/s^2
velocities = [np.sqrt(2*g*x/100) for x in heights] 
print ("v0 = {:.2f} ± {} m/s".format(np.mean(velocities), uncertainty))

v0 = 5.06 ± 0.08 m/s


##### (b) Set the launch angle to $35^\circ$. Record an uncertainty for this number.

We set the launch angle to $35.0^\circ \pm 0.5^\circ$.

##### (c) Use $g = 9.81 \pm .05 \frac{m}{s^2}$

##### (d) Take three partial derivatives of function 4.16: $\frac{\delta f}{\delta g}$, $\frac{\delta f}{\delta \theta}$, $\frac{\delta f}{\delta v_0}$ and record them in your lab notebook. Evaluate these functions at the measurement values.

$$
\frac{\delta f}{\delta g} = \frac{\delta}{\delta g}\bigg(\frac{v_0^2 \sin(2\theta)}{g}\bigg) 
     = - \frac{v_0^2 \sin(2\theta)}{g^2}
$$
<br>
$$
\frac{\delta f}{\delta g} \bigg(v_0=5.06, g=9.81, \theta = 35.0\bigg) = - \frac{(5.06)^2 \sin(2(35.0))}{9.81^2} = - 0.250
$$
<br>
<br>


$$
\frac{\delta f}{\delta v_0} = \frac{\delta}{\delta v_0}\bigg(\frac{v_0^2 \sin(2\theta)}{g}\bigg) 
     = \frac{2v_0 \sin(2\theta)}{g}
$$
<br>
$$
\frac{\delta f}{\delta v_0} \bigg(v_0=5.06, g=9.81, \theta = 35.0\bigg)=  \frac{2(5.06) \sin(2(35.0))}{9.81} = 0.969
$$
<br>
<br>


$$
\frac{\delta f}{\delta \theta} = \frac{\delta}{\delta \theta}\bigg(\frac{v_0^2 \sin(2\theta)}{g}\bigg) 
     = \frac{2v_0^2 \cos(2\theta)}{g}
$$
<br>
$$
\frac{\delta f}{\delta \theta} \bigg(v_0=5.06, g=9.81, \theta = 35.0\bigg)= \frac{2(5.06)^2 \cos(2(35.0))}{9.81} = 1.785^\circ, 0.0137 rad
$$
<br>
<br>


##### (e) Apply equation 4.9 to determine the uncertainty in the range of your projectile.

<sub>4.9</sub>
$$
\Delta f = \sqrt{\bigg(\frac{\Delta f}{\Delta g} \Delta g \bigg)^2 
               + \bigg(\frac{\Delta f}{\Delta v_0} \Delta v_0 \bigg)^2 
               + \bigg(\frac{\Delta f}{\Delta \theta} \Delta \theta \bigg)^2}
$$
<br>
$$
\Delta f = \sqrt{\big(-0.250 * 0.05 \big)^2 
               + \big(0.969 * 0.08 \big)^2 
               + \big(0.0137 * 0.0087\big)^2} = 0.0785m
$$
<br>


##### (f) Launch the projectile and observe whether it lands in the uncertainty window. Record your results.

We predict that the range is: 
$$
x = \frac{v_0^2 \sin(2\theta)}{g} = \frac{(5.06 \frac{m}{s})^2 \sin(2(35^\circ))}{9.81\frac{m}{s^2}} = 2.452m
$$

With the calculated uncertainty: $2.452m \pm 0.0785m$ We launched the rocket, and it right on 2.44m. This was definately in the uncertainty window!


## P4.6
You may be familiar with a simple pendulum if you’ve ever owned a grandfather
clock. The swinging mass is a pendulum and it’s period can be
calculated like this

$$
T = 2\pi \sqrt{\frac{L}{g}}
$$

where $L$ is the length of the pendulum, $T$ is the period, and $g$ is the acceleration
due to gravity. Instead of using this equation to calculate the
period, we can measure a pendulum’s period and use the above expression
to determine a value for the acceleration due to gravity.




##### (a) Measure the length of the pendulum and attach an uncertainty to that measurement.

Length = $1.25m \pm 0.001$

##### (b) Use the photogate to measure the period of the pendulum and record it’s uncertainty.

| Trial # | Time (in seconds)|
|--|
| 1 | $2.2551 \pm 0.0001$ |
| 2 | $2.2707 \pm 0.0001$ |
| 3 | $2.2601 \pm 0.0001$ |
| 4 | $2.2672 \pm 0.0001$ |

The mean being: $2.2633s  \pm 0.0001s$

##### (c) Solve equation 4.17 for $g$

$$
g = \frac{L}{\big(\frac{T}{2\pi}\big)^2} 
$$

##### (d) Plug your measurements into the expression for $g$ to calculate it’s value.

$$
g = \frac{L}{\big(\frac{T}{2\pi}\big)^2} = \frac{1.25m}{\big(\frac{2.2633s}{2\pi}\big)^2} = 9.63\frac{m}{s^2}
$$

##### (e) Take two partial derivatives of the function for g that you got above:

$$
\frac{\delta g}{\delta L} = \frac{\delta}{\delta L}\Bigg(\frac{L}{\big(\frac{T}{2\pi}\big)^2} \Bigg)
      = \frac{1}{\big(\frac{T}{2\pi}\big)^2}
$$
<br>
<br>

```
DUE TO TIME CONSTRAINTs, I WAS UNABLE TO DO THE REST OF THE LAB; MISSING HALF OF THIS PROBLEM AND THE NEXT PROBLEM. I FEEL COMFORTABLE WITH THE SUBJECT THOUGH, AS P4.6 IS ESSENTIALLY A REPEAT of P4.5.
```

# Take home thoughts

I think the lab was helpful to getting back in the groove of calculus. It's been a couple years sice I took MATH 316. I will admit, I hated typing this lab up. LaTex is not the most friendly thing to do math in, only to write it up. I think that is part of the reason why I didn't finish. But I am proud of how clean my lab notebook looks.

# Sign off
 - Kyle West
 - Brandon Welch